From 2aae5d2280d02812669ac38e3981692b98de7c10 Mon Sep 17 00:00:00 2001
From: Dor Askayo <dor.askayo@gmail.com>
Date: Sat, 3 Apr 2021 16:37:30 +0300
Subject: [PATCH] graphene-config: Enable NEON for AArch64 on Linux

NEON is fully supported by AArch64. However, GCC doesn't seem to define
__ARM_NEON__ or _M_ARM64 for AArch64.

Using __aarch64__ should allow a proper detection of this case.
---
 include/graphene-config.h.meson | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/graphene-config.h.meson b/include/graphene-config.h.meson
index 949eee7..96192cc 100644
--- a/include/graphene-config.h.meson
+++ b/include/graphene-config.h.meson
@@ -19,7 +19,7 @@ extern "C" {
 #mesondefine GRAPHENE_HAS_SSE
 # endif
 
-#  if defined(__ARM_NEON__) || defined (_M_ARM64)
+#  if defined(__ARM_NEON__) || defined (_M_ARM64) || defined (__aarch64__)
 #mesondefine GRAPHENE_HAS_ARM_NEON
 #  endif
 
